packages/headless/src/controllers/triggers/headless-execute-trigger.ts

```diff
@@ -0,0 +1,90 @@
+import {TriggerSection} from '../../state/state-sections';
+import {triggers} from '../../app/reducers';
+import {buildController, Controller} from '../controller/headless-controller';
+import {loadReducerError} from '../../utils/errors';
+import {logTriggerExecute} from '../../features/triggers/trigger-analytics-actions';
+import {SearchEngine} from '../../app/search-engine/search-engine';
+
+/**
+ * The `ExecuteTrigger` controller handles execute trigger actions.
+ */
+export interface ExecuteTrigger extends Controller {
+  /**
+   * the state of the `ExecuteTrigger` controller.
+   */
+  state: ExecuteTriggerState;
+}
+
+/**
+ * A scoped and simplified part of the headless state that is relevant to the `ExecuteTrigger` controller.
+ */
+export interface ExecuteTriggerState {
+  /**
+   * The name of the function to be executed.
+   */
+  name: string;
+
+  /**
+   * The parameters of the function to be executed.
+   */
+  params: {}[];
```

Comment:
Should the array type be preceded by an object?

It seems in the docs the type is `string | number | boolean`. Are non-primitives like objects and arrays supported?

Reply:
I put an object because that is what it was originally defined as in the `triggers.ts` file but I think you're correct and should change it to that.